tests/models/zoedepth/test_modeling_zoedepth.py

```diff
@@ -255,3 +280,78 @@ def test_inference_depth_estimation_multiple_heads(self):
         ).to(torch_device)
 
         self.assertTrue(torch.allclose(outputs.predicted_depth[0, :3, :3], expected_slice, atol=1e-4))
+
+    def check_target_size(
+        self,
+        image_processor,
+        pad_input,
+        images,
+        outputs,
+        raw_outputs,
+        raw_outputs_flip=None,
+    ):
+        outputs_large = image_processor.post_process_depth_estimation(
+            raw_outputs,
+            [img.size[::-1] for img in images],
+            outputs_flip=raw_outputs_flip,
+            target_sizes=[tuple(np.array(img.size[::-1]) * 2) for img in images],
+            do_remove_padding=pad_input,
+        )
+
+        for img, out, out_l in zip(images, outputs, outputs_large):
+            out_l_reduced = torch.nn.functional.interpolate(
+                out_l.unsqueeze(0).unsqueeze(1), size=img.size[::-1], mode="bicubic", align_corners=False
+            )
+            self.assertTrue((np.array(out_l.shape)[::-1] == np.array(img.size) * 2).all())
+            self.assertTrue(torch.allclose(out, out_l_reduced, rtol=1e-2))
+
+    def check_post_processing_test(self, image_processor, images, model, pad_input=True, flip_aug=True):
+        inputs = image_processor(images=images, return_tensors="pt", do_pad=pad_input).to(torch_device)
+
+        with torch.no_grad():
+            raw_outputs = model(**inputs)
+            raw_outputs_flip = None
+            if flip_aug:
+                raw_outputs_flip = model(pixel_values=torch.flip(inputs.pixel_values, dims=[3]))
+
+        outputs = image_processor.post_process_depth_estimation(
+            raw_outputs,
+            [img.size[::-1] for img in images],
+            outputs_flip=raw_outputs_flip,
+            do_remove_padding=pad_input,
+        )
+
+        expected_slice = torch.tensor(self.expected_slice_post_processing[pad_input, flip_aug]).to(torch_device)
+        for img, out in zip(images, outputs):
+            self.assertTrue(img.size == out.shape[::-1])
+            self.assertTrue(torch.allclose(expected_slice, out[:3, :3], rtol=1e-3))
+
+        self.check_target_size(image_processor, pad_input, images, outputs, raw_outputs, raw_outputs_flip)
+
+    def test_post_processing_depth_estimation_post_processing_nopad_noflip(self):
+        images = [prepare_img()] * 2
```

Comment:
The best test here would to have two input images of different sizes

Reply:
Zoedepth preprocessing does not seem to support batching together different-sized images

Reply:
It should - the image processor resizes the individual images to `size["height"], size["width"]`. What is the error you're hitting?

Reply:
Hey, sorry for the delay here, I missed the notification.
So, the problem is inside the image processor and has nothing to do with the changes in this PR. What happened is that inside the processor, there is a flag `keep_aspect_ration` that is set to true by default, thus the resizing ends up with non-equal-sized images. I changed the flag for the tests and it works now.